src/batch-client/index.ts

```diff
@@ -1335,16 +1335,27 @@ export class ZimbraBatchClient {
 			singleRequest: true
 		});
 
-	public modifyPrefs = (prefs: PreferencesInput) =>
-		this.jsonRequest({
+	public modifyPrefs = (prefs: PreferencesInput) => {
+
+		let attrs: any = mapValuesDeep(prefs, coerceStringToBoolean);
+		const {zimbraPrefMailTrustedSenderList} = prefs;
+
+		if (zimbraPrefMailTrustedSenderList &&
```

Comment:
this needs to be handled in a generic way for all array type prefs

Reply:
sure i will add code for generic way